server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -99,7 +128,13 @@ public void parse(ParseContext context) throws IOException {
     @Override
     protected void parseCreateField(ParseContext context) throws IOException {
         for (String field : context.getIgnoredFields()) {
-            context.doc().add(new Field(NAME, field, Defaults.FIELD_TYPE));
+            if (!fieldType().hasDocValues()) {
+                // Use legacy field type if there are no doc values
+                context.doc().add(new Field(NAME, field, Defaults.LEGACY_FIELD_TYPE));
```

Comment:
We still want to index the field so we should always add the `Field` but switch to the right `FIELD_TYPE` depending on the created version.

Reply:
I'm not sure if I need FIELD_TYPE for this, because I added the SortedDocValues
https://github.com/elastic/elasticsearch/pull/60103/files#diff-374f61a69e82c881c670a6df0774cd4bR135
```java
    if (fieldType().hasDocValues()) {
        final BytesRef binaryValue = new BytesRef(field);
        context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
    }
```